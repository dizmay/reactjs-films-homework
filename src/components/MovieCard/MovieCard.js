import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './MovieCard.scss';
import { genresSelector } from '../../utils/selectors';
import MovieTrailer from '../MovieTrailer/MovieTrailer';
import MovieCardHover from '../MovieCardEffects/MovieCardHover';
import MovieInfo from '../MovieInfo/MovieInfo';

const MovieCard = ({
  title,
  rate,
  poster,
  genres,
  movieGenres,
  overview,
  movieId,
  movieTrailer,
  getMovieTrailer,
}) => {
  const [hover, setHover] = useState(false);
  const [trailer, setTrailer] = useState(false);
  const showTrailer = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
    setTrailer(!trailer);
    getMovieTrailer(movieId);
  };
  const showHover = () => {
    setHover(true);
  };
  const hideHover = () => {
    setHover(false);
  };
  return (
    <div className={styles.card}>
      {
        trailer && <MovieTrailer showTrailer={showTrailer} movieTrailer={movieTrailer} />
      }
      <div className={styles.card__image} onMouseEnter={showHover} onMouseLeave={hideHover}>
        <MovieCardHover
          hover={hover}
          overview={overview}
          title={title}
          rate={rate}
          genres={movieGenres.length > 0 ? genresSelector(movieGenres, genres) : 'No genres'}
          showTrailer={showTrailer}
        />
        <img src={poster ? `https://image.tmdb.org/t/p/original${poster}` : 'http://underscoremusic.co.uk/site/wp-content/uploads/2014/05/no-poster.jpg'} alt="Movie poster" />
      </div>
      <div className={styles.card__details}>
        <MovieInfo black title={title} rate={rate} genres={movieGenres.length > 0 ? genresSelector(movieGenres, genres) : 'No genres'} />
      </div>
    </div>
  );
};

MovieCard.defaultProps = {
  title: '',
  rate: 0,
  poster: '',
  genres: [],
  movieGenres: [],
  overview: '',
  movieId: 0,
  movieTrailer: '',
  getMovieTrailer: null,
};

MovieCard.propTypes = {
  title: PropTypes.string,
  rate: PropTypes.number,
  poster: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.object),
  movieGenres: PropTypes.arrayOf(PropTypes.number),
  overview: PropTypes.string,
  movieId: PropTypes.number,
  movieTrailer: PropTypes.string,
  getMovieTrailer: PropTypes.func,
};

export default MovieCard;
