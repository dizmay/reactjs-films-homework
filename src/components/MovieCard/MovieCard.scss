* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: min-content;
  width: 20vw;
  background: #fff;
  box-shadow: 0 10px 20px -5px #000;
}

.card__image {
  width: 100%;
  height: 80%;
  position: relative;
}

.card__image img {
  width: 100%;
  height: 100%;
  object-fit: fill;
}

.card__details {
  width: 100%;
  height: 30%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  padding: 1rem;
}
