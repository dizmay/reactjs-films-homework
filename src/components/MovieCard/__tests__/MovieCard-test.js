import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from '../MovieCard';

it('renders correctly', () => {
  const component = renderer.create(<MovieCard />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
