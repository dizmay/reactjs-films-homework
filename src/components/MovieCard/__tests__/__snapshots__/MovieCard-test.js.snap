// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card__image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="hover_hideContainer"
    >
      <div
        className="hover__watchNow"
      >
        <button
          className="watchNow__btn"
          onClick={[Function]}
          type="button"
        >
          â–º
        </button>
        <span
          className="watchNow__title"
        >
          Watch Now
        </span>
      </div>
      <button
        className="btn transparent"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <img
      alt="Movie poster"
      src="http://underscoremusic.co.uk/site/wp-content/uploads/2014/05/no-poster.jpg"
    />
  </div>
  <div
    className="card__details"
  >
    <div
      className="info__container"
    >
      <div
        className="info"
      >
        <span
          className="info__title info__title_black"
        >
          
        </span>
        <div
          className="info__rate"
        >
          0.0
        </div>
      </div>
      <div
        className="info__genres"
      >
        No genres
      </div>
    </div>
  </div>
</div>
`;
